### PR TITLE
Allow selecting which files to publish

### PR DESCRIPTION
**Pull request by danielrbradley**
Allow the use of this action to customise which files from the source directory to copy and publish.

Conversation for context: https://github.com/pulumi/ci-mgmt/pull/920#issuecomment-2107887622

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -13,6 +13,9 @@ inputs:
   source:
     description: Path to the directory containing the Go source to be published. This folder should contain at least a go.mod file.
     required: true
+  files:
+    description: List of glob patterns to select Go source to be published. Defaults to "**" (all files).
+    default: "**"
```
Comment (flostadler) [inline action.yml:18]: If we're only caring about go sources here, why don't we use `["*.go", "go.mod", "go.sum"]` as a default?
Wondering if we could end up publishing unwanted changes with this default

**Review by flostadler — APPROVED** [review]
Just one small question

File: action.yml (hunk 1 above)
Comment (danielrbradley) [reply action.yml:18]: I'm planning to make this the default value we set via ci-mgmt, but I think it's least suprising for this action to just publish everything in the selected folder by default if you've not asked to filter anything out.

Comment (flostadler) [reply action.yml:18]: I guess you're right! It's probably more surprising if we run into "why are those files not published" issues

**Pull request merged**